visualization/include/pcl/visualization/impl/pcl_visualizer.hpp

```diff
@@ -672,23 +672,45 @@ pcl::visualization::PCLVisualizer::addText3D (
   if (viewport < 0)
     return false;
 
-  // check all or an individual viewport for a similar id
+  // If there is no custom viewport and the viewport number is not 0, exit
+  if ((rens_->GetNumberOfItems () == 1) && (1 <= viewport))
```

Comment:
Wouldn't 
```cpp
if (viewport => rens_->GetNumberOfItems ())
```
be generic for all cases and actually accomplish the same ?

Reply:
Ok.